### DISCUSSION TITLE
Wrong end time or event duration?

### DISCUSSION THREAD

**Discussion by DananFlander** [derekantrican/GAS-ICS-Sync#464] [Q&A] [answered]
I have events syncing, but either the duration isn't copying over correctly or perhaps the end time isn't? Events copy over as 0 minute duration or same start/end time.

![image](https://github.com/user-attachments/assets/be518c7e-f560-4161-9f6a-2d51a8b3d5fd)

Orange is the original ics link, brown is the merged calendar. All other information (title, description, link) merge over correctly.

Is there a setting I missed before the install or a toggle somewhere to correct this?

**Comment by jonas0b1011001** [comment]
> Is there a setting I missed before the install or a toggle somewhere to correct this?

No.
Can you share one of the event definitions? (download ics, open in text editor, search for the event title, share `BEGIN:VEVENT` to `END:VEVENT`)

**Comment by DananFlander** [comment]
Here's the entire ICS instead:

```BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//PRACC.COM//Golang ICS Library
METHOD:REQUEST
DESCRIPTION:Upcoming matches for team Undone
REFRESH-INTERVAL;VALUE=DURATION:PT10M
BEGIN:VEVENT
UID:match-2318229@pracc.com
CREATED:20241111T003628Z
DTSTAMP:20241111T013117Z
LAST-MODIFIED:20241111T003628Z
DTSTART:20241111T040000Z
SUMMARY:vs. Holly Molly (Map: de_nuke)
DESCRIPTION:Server: 169.150.232.56:26913\, Password: -- hidden --\,
  Provided by: Undone
URL:https://pracc.com/matches/2318229
END:VEVENT
BEGIN:VEVENT
UID:match-2318237@pracc.com
CREATED:20241111T010008Z
DTSTAMP:20241111T013117Z
LAST-MODIFIED:20241111T010008Z
DTSTART:20241111T030000Z
SUMMARY:vs. Final Form (Map: de_nuke)
DESCRIPTION:Server: 169.150.232.56:26913\, Password: -- hidden --\,
  Provided by: Undone
URL:https://pracc.com/matches/2318237
END:VEVENT
END:VCALENDAR
```

**Reply by jonas0b1011001** [reply, marked as answer]
The file is missing an end date/event duration, 60min seems to be a default value used by google.

If this is important for your use case, we can of course adapt this default.

**Reply by DananFlander** [reply]
Ah, makes sense. It's not important; mostly just an eyesore and I'll be using the script regardless.

**Reply by jonas0b1011001** [reply]
If you want to change it, replace
https://github.com/derekantrican/GAS-ICS-Sync/blob/dd4c000080fe392fa893adb9ed5e977dbc1d0bbd/ical.js.gs#L8958-L8971
with
```
      key: "endDate",
      get: function get() {
        var endDate = this._firstProp('dtend');
        if (!endDate) {
          var duration = this._firstProp('duration');
          endDate = this.startDate.clone();
          if (duration) {
            endDate.addDuration(duration);
          } else if (endDate.isDate) {
            endDate.day += 1;
          } else {
            endate = endDate.adjust(0,1,0,0)
          }
        }
        return endDate;
      },
```

Sync to a new calendar for the events to update.

**Reply by DananFlander** [reply]
Works perfect. Thanks!
